calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -235,40 +256,63 @@ func (w *PeerWatcher) resyncAndCreateWatcher(state map[string]*bgpPeer) error {
 			for ip, asn := range ipAsn {
 				existing, ok := state[ip]
 				if ok {
+					w.log.Debugf("peer(update) neighbor ip=%s for BGPPeer=%s", ip, peer.ObjectMeta.Name)
 					existing.SweepFlag = false
-					if existing.AS != asn {
-						existing.AS = asn
-						err := w.updateBGPPeer(ip, asn)
+					oldSecret := ""
+					if existing.BGPPeerSpec.Password != nil && existing.BGPPeerSpec.Password.SecretKeyRef != nil {
+						oldSecret = existing.BGPPeerSpec.Password.SecretKeyRef.Name
+					}
```

Comment:
We should probably have a function for this logic, as we're doing it quite a few times

Reply:
Agree